frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -2720,7 +2720,8 @@ bool TypeInference::hasVarbitsOrUnions(const IR::Type* type) const {
 
 bool TypeInference::onlyBitsOrBitStructs(const IR::Type* type) const {
     // called for a canonical type
-    if (type->is<IR::Type_Bits>()) {
+    if (type->is<IR::Type_Bits>() || type->is<IR::Type_InfInt>() ||
```

Comment:
IR::Type_InfInt shouldn't be allowed here, we can only have fixed width integers (represented by IR::Type_Bits).

Reply:
Good catch.  Sorry, I missed that Type_InfInt does not represent ```int<W>```.